### ISSUE TITLE
DirectAssociationNotConfigured for class with multiple polymorphic relationships

### ISSUE THREAD

**Issue by foxpowa** [605data/strongbolt#39]
Hi!
It seems we have a great gem here !
Unfortunatly I can't add it into my Rails app.
Trying to run `rails c` or `rails s`, I'm getting this error :

> strongbolt-0.3.12/lib/strongbolt/tenantable.rb:92:in `setup_model': Class PublicActivity::Activity is 1 degree from Project but the association isn't configured, you should implement it before using tenant method (Strongbolt::DirectAssociationNotConfigured)

Going deeper into your gem, I found that for setup a tenant (in my case Project class) Strongbolt compute the inverse relation.

```
module Strongbolt
  module Tenantable
    module ClassMethods
    ...
    def setup_model
       ...
       inverse = inverse_of(association)
       ...
    end
  end
end
``` 

However, I'm using a gem, [public activity](https://github.com/chaps-io/public_activity), which define an Activity class. It has 3 polymorphic associations (trackable, owner, recipient).
Project records are used through trackable relationship but there is no way to tell to Strongbolt.

Is there something I missed (configuration ...) ?

**Comment by tobithiel** [comment]
Yeah with multiple polymorphic associations on the same model, Strongbolt has troubles finding the inverse. The relevant code is here: https://github.com/605data/strongbolt/blob/master/lib/strongbolt/tenantable.rb#L257 and as you see with multiple polymorphic associations, it won't be able to guess correctly.

You can try adding `, inverse_of: ...` to the association, if that's unique on one of the sides. While it is strictly speaking not supported on polymorphic associations with ActiveRecord, I don't think ActiveRecord will complain and Strongbolt will use that help to find the inverse anyways.

Another approach would be to make the `inverse_of` method more intelligent and look at the other association attributes, but out-of-the-box, you only have the option above.

**Comment by foxpowa** [comment]
Thanks @tobithiel  that what I saw in tenantable.rb ;)
Here is my version of `inverse_of` function I tested

```
      def inverse_of association
        # If specified in association configuration
        return association.inverse_of if association.has_inverse?

        polymorphic_associations = []

        # Else we need to find it, using the class as reference
        association.klass.reflect_on_all_associations.each do |assoc|
          # If the association is polymorphic
          if assoc.options.has_key? :polymorphic
            polymorphic_associations << assoc
            if association.options and association.options[:as] == assoc.name
              return assoc
            end

          # If same class than the original source of the association
          elsif assoc.klass == association.active_record

            Strongbolt.logger.debug "Selected inverse of #{association.name} between #{association.active_record} " +
              "and #{association.klass} is #{assoc.name}.\n " +
              "If not, please configure manually the inverse of #{association.name}\n"

            return assoc
          end
        end

        if polymorphic_associations.size == 1
          return polymorphic_associations.first
        end

        return nil
      end
```

I kept polymorphic_assoc array in case of assocaition.options[:as] was not defined or whatever.
Tell me if you want a PR ;)

**Comment by tobithiel** [comment]
@foxpowa That looks great. If you open a PR that would be great 👍

**Comment by tobithiel** [comment]
Resolved through #40
